### ISSUE TITLE
Very slow compilation, even on figwheel reload

### ISSUE THREAD

**Issue by conan** [plexus/chestnut#114]
I've recently switched from OSX to Windows, and find my chestnut projects build very slowly (~4-5 minutes).  This is true even in a Ubuntu vagrant box, but on OSX it used to take around 30 seconds.  This also occurs when I'm running a figwheel repl, every time I make a change, I have to wait the same length of time as the initial build (the full 4-5 minutes, instead of the 1-2 seconds I expect).  

Here are the relevant bits of my project.clj:

```
:plugins [[lein-cljsbuild "1.0.3"]]
:profiles {:dev {:dependencies [[figwheel "0.2.1-SNAPSHOT"]
                                [figwheel-sidecar "0.2.1-SNAPSHOT"]
                                [com.cemerick/piggieback "0.1.3"]
                                [secretary "1.2.1"]
                                [weasel "0.4.2"]]
                 :plugins      [[lein-figwheel "0.2.1-SNAPSHOT"]]
                 :repl-options {:init-ns my-project.server
                                :nrepl-middleware [cemerick.piggieback/wrap-cljs-repl]}}}
```
I'm wondering whether it's a problem with Windows notifying of filesystem changes, but that doesn't explain the very long overall build time; it's quicker for my colleagues on OSX, and it was quicker for me on the same hardware when I was running OSX instead of Windows or Ubuntu.

**Update:** Whilst writing this I tried copying the project to a folder in my Ubuntu VM that was not shared with Windows, and it's much faster - the speed I expect.  This appears to be a problem with the Windows filesystem, but I'm not sure how to investigate further.  The problem is that I can't use my tools in Windows to work on the files if the filesystem is not shared.

**Comment by conan** [comment]
Is it possible to run this stuff on a Windows filesystem at all?  I tried sharing my linux VM's filesystem with my host Windows over samba and running a project there, but it's still slow.  It seems it's only feasible speed on OSX or linux filesystems.

**Comment by conan** [comment]
I've tried turning off my antivirus and disabling indexing, two classic causes of slow filesystem access on Windows, but sadly no joy.

**Comment by conan** [comment]
Has anyone else had success using a chestnut project on Windows?  I'd be interested to know what it looks like.

**Comment by conan** [comment]
I've done some inspecting of files that are open in the java processes, and it looks like a lot of time is spent reading `core.cljs` and writing `core.js`.  I understand the reading, but on an incremental compile, surely there's no need to write `core.js` at all?

**Comment by conan** [comment]
I've confirmed that the problem is that all files are being recompiled whenever a single change is made.  I've tried running the following watchtower snippet to confirm that it's not a problem in watchtower:
```
(watcher ["src/" "cljs/"] 
  (rate 50) 
  (file-filter (extensions :cljs)) 
  (on-change #(println "files changed: " %)))
```
I ran this in a clean chestnut project REPL, and added a second namespace alongside core.cljs.  Changes to either of the namespace only reported the change I made, so it appears watchtower is reporting correctly (in my artificial test).

**Comment by conan** [comment]
I've confirmed that compilation is fast when running `lein figwheel` instead of using chestnut's `(run)`.  Is this something to do with enlive's `reload/watch-service` function perhaps?

**Comment by plexus** [comment]
@bhauman do you have any idea about this? could it be a difference between figwheel-sidecar vs lein-figwheel?

**Comment by bhauman** [comment]
It's really hard to say from this information.  Need a smaller reproducible example. Lots of things going on here.

**Comment by bhauman** [comment]
I would also look at the different windows issues that have been patched since 0.2.1
Like this one:
https://github.com/bhauman/lein-figwheel/issues/73

**Comment by conan** [comment]
To reproduce: 
```
lein new chestnut speed-test
cd speed-test
lein repl
(run)
```
Wait for initial compilation to complete, then modify a file.  If the problem is occurring, it'll take a similar order of magnitude of time to finish compiling the changes, if it's not occurring it'll happen in under a second.

**Comment by conan** [comment]
I've tested on Windows 8.1 on a Macbook Pro and a Lenovo Yoga 2 Pro, in Windows 10 Technical Preview in a VM on the Lenovo, and also in Ubuntu in a VM on the Mac.  I was able to reproduce in all cases, including Ubuntu, but the problem only occurred on Ubuntu when the files were being shared from the host (Windows) operating system with the guest (Linux) - i.e. they were on a Windows disk.  If I move the files somewhere else in the VM that wasn't shared, everything is fine.
